tools/clang/test/HLSLFileCheck/passes/llvm/simplifycfg/fold-cond-branch-on-phi.hlsl

```diff
@@ -7,6 +7,7 @@
 // CHECK: %[[cond:.+]] = phi i1
 // CHECK-SAME: [ false
 // CHECK: br i1 %[[cond]]
+// CHECK: @main
```

Comment:
I see one added check: for `@main` after the checks for things tat should be *in* main, which is strange.  This test uses /Zi, so it should include the special CHECK-LABEL after IR checks to make sure you're not matching strings from the included source file:
```
// Exclude quoted source file (see readme)
// CHECK-LABEL: {{!"[^"]*\\0A[^"]*"}}
```

Reply:
As it happens, the previous checks won't match the debug inclusions because they include regexp, which is why this shader didn't have it before. I don't know why ralf added the `@main` check, but it's often just a check that nothing failed and the debug info that would match `@main` won't be produced unless a successful compile has occurred. 

Adding the suggested CHECK-LABEL might be a good precaution regardless, but the readme it references has been removed so it's probably more confusing to reference it than not.

Reply:
The readme that was removed used to be hard to find, as it was nested under some debug sub-folder, and there were other readme files that weren't the one you were looking for.  Since then, the text explaining this was moved into the top-level Readme.md under the HLSLFileCheck directory, so it should be much easier to find.

I too do not know why this `@main` check was added here in the first place.  My worry was that the @main check might only match because it was matching a sub-string in the included file string, since it should be located before the instructions in the function, right?

Reply:
Good to know! I thought it was just gone.

I share your worry and await consolation.

Reply:
Damn it... I overlooked this one, failed to properly revert it :(.
It was only added previously when I had the rest of the checks disabled because tests without any CHECK fail.

Reply:
No worries. If you want to make a quick fix change, I promise it won't take as long as this one did 😉

Reply:
No problem, but I need some advice how to submit the fix. A new PR? Or simply push a new commit to this one and you guys will merge the lifetimes branch again?

Reply:
A new PR would be simplest

Reply:
Here you go :).
https://github.com/microsoft/DirectXShaderCompiler/pull/3264
For future reference, could I have reused the same 'lifetimes' branch on my fork instead of creating a new branch? I was unsure, so went for the "safe" variant.

Reply:
Once this PR was completed, you could use the same branch to create another PR.